### ISSUE TITLE
Error handling in "pdnsutil check zone"

### ISSUE THREAD

**Issue by krionbsd** [PowerDNS/pdns#6151]
- Program: Authoritative
 - Issue type: Bug report

### Short description

pdnsutil check zone should check only given zone 

### Environment

 - Operating system:  FreeBSD 11.1-RELEASE
 - Software version: PowerDNS Auth 4.1.0 + DNSSEC

### Steps to reproduce
1) Generate any random 256 char name:
cat /dev/urandom | env LC_CTYPE=C tr -cd 'a-f0-9' | head -c 256
2) Update zone file:
echo "0e8f032ef4efae5740248de6c69b30c2b15ef98b8647cbb824a64e129e20add35cab929ea20191d69e40eb859f0adf0b49e1b7229dd1a3991981e28d3c0ad74cb6d87d2b5af686196815d5f5868ab9a0590269e890cdc5f4a4d262d1d8b9023c8f0a46206fc138b65c2474f79c0cd9ff4743356a666670d2f355fe79321c3832 IN A IPv4" >> krion.cc.zone
3) Error message "Error: name too long to append" appears in /var/log/messages
4) dns:/usr/local/etc/pdns/zones # pdnsutil check-zone krion.cc
Jan 05 20:49:41 Caught an exception instantiating a backend, cleaning up
Error: name too long to append
dns:/usr/local/etc/pdns/zones # pdnsutil check-zone bsdop.net
Jan 05 20:49:43 Caught an exception instantiating a backend, cleaning up
Error: name too long to append

pdnsutil checks bsdop.net and produces error for krion.cc zone.

### Expected behaviour
"pdnsutil check zone" checks given zone only

**Comment by Habbie** [comment]
pdnsutil is aborting while instantiating the bindbackend (which loads all zones), before it even gets to actually checking the zone.

**Comment by krionbsd** [comment]
Can zone be checked regardless of backend or can some argument be added to omit backend init?

**Comment by Habbie** [comment]
> Can zone be checked regardless of backend or can some argument be added to omit backend init?

I recall getting this request before but I cannot find a ticket for it right now. As a compromise, perhaps we could at least add the zone name to the error!

**Comment by krionbsd** [comment]
Agree, having zone name in error message would make sense as well 👍

**Comment by miodvallat** [comment]
Current status:
- the bind backend no longer fails to load with invalid data, but will not serve the zone in error
- the error message can be made sense of:
```
Jun 16 12:11:53 [bindbackend] Parsing 1 domain(s), will report when done
Jun 16 12:11:53 [bindbackend] error at 2025-06-16 12:11:53 +0200 parsing 'example.com' from file 'zones/example.com': label too long to append: ultralonglabelwhichspansmorethansixtythreecharactersbecausewearerebels
Jun 16 12:11:53 [bindbackend] Done parsing domains, 1 rejected, 1 new, 0 removed
```
- `pdnsutil list-zone` will fail with a meaningful error message too:
```
Error: zone was not loaded, perhaps because of:  error at 2025-06-16 12:15:30 +0200 parsing 'example.com' from file 'zones/example.com': label too long to append: ultralonglabelwhichspansmorethansixtythreecharactersbecausewearerebels
```
- `pdnsutil check-zone` will not work:
```
[Error] SOA lookup failed for zone 'example.com': Zone for 'example.com' in 'zones/example.com' not loaded (file missing, corrupt or primary dead)
```
